src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/RecordComponentNumberCheck.java

```diff
@@ -156,11 +154,14 @@ public void visitToken(DetailAST ast) {
      * @return the number of record components in this record definition
      */
     private static int countComponents(DetailAST recordComponents) {
-        final AtomicInteger count = new AtomicInteger(0);
-        TokenUtil.forEachChild(recordComponents,
-            TokenTypes.RECORD_COMPONENT_DEF,
-            node -> count.getAndIncrement());
-        return count.get();
+        int count = 0;
+        for (DetailAST child = recordComponents.getFirstChild();
+             child != null; child = child.getNextSibling()) {
+            if (child.getType() == TokenTypes.RECORD_COMPONENT_DEF) {
+                count++;
```

Comment:
please reuse this
https://github.com/checkstyle/checkstyle/blob/614c3f452b326776b6785b3fcddc32b0091bb24e/src/main/java/com/puppycrawl/tools/checkstyle/DetailAstImpl.java#L194-L202